src/util/compareThought.ts

```diff
@@ -84,10 +94,14 @@ export const makeOrderedComparator = (comparators: ComparatorFunction<any>[]): C
       // base case
       ? 0
       // if the values are non equal by the initial comparator, return the result of the comparator
-      : comparators[0](a, b) ||
+      : execComparator(comparators[0], a, b) ||
         // if they are equal, move on to the next comparator
         makeOrderedComparator(comparators.slice(1))(a, b) // RECURSION
 
+/** Executes and comparator with respective string modifications. */
+const execComparator = (comparator: ComparatorFunction<any>, a:any, b: any) =>
+  comparator.name === 'compareStringsWithEmojis' ? comparator(a, b) : comparator(removeEmojisAndSpaces(a), removeEmojisAndSpaces(b))
```

Comment:
This is very creative, but unfortunately couples the wrong things together. `execComparator` sounds like a general function, but is tightly coupled to `compareStringsWithEmojis` and `removeEmojisAndSpaces`. Why would a general function have specialized knowledge about emojis? Further, `makeOrderedComparator` depends on `execComparator`, inheriting all of those same couplings, defeating the purpose of a generalized `makeComparator` function.

I will see if I can come up with a preferable solution so you can see what I mean.